docs/architecture.md

```diff
@@ -0,0 +1,122 @@
+
+The coordinator will have to expose an API along the p2p API for the lightning (including dlc) communication e.g. order book. **Assumption**: 10101 will not exclusively talk via the p2p protocol with the app. As DLCs should get integrated into upstream rust-lightning a more generalised approach would be beneficiary.  
+
+- **Option 1 *Restfull API***: Simple API to expose CRUD operations through an HTTP. Broadly used and well understood.
+- **Option 2 *Websocket***: Only one API, exposing all sorts of data the client can subscribe to.
+
+A combination of both options seem to be reasonable as the approach in itchysats with only web sockets seemed artificial and introduced unneeded complexities. However using websockets to push updates to subscribed data objects (instead of polling) seems to be a good practice for multiple reasons (bandwidth, freshness, etc.).
+
+*We also shortly discussed graphql, but probably figured it to be not beneficiary (as too powerful and complex) for our use case.*
+  
+### RPC vs REST: 
+
+It seems RPC is broadly used nowadays. We might want to consider evaluating RPC for our Coordinator APIs.
+
+## Security
+
+One of our most important features is Security. We must not only think of it from a point of view that the protocol is secure. Equally we need to look at the way we derive and manage crypto materials. 
+
+### How to secure the cryptographic key on the phone?
+
+- **Option 1 *HSM***: iPhone and Android support hardware security modules on the phone. For iPhone its Secure Element - We could use such module (isolated from the os) to securily perfom cryptographic algorithms without ever exposing the private key. *Note, its unclear what that would mean for exporting the key if required* 
+- **Option 2 *Encrypt keys on disk***: Simply encrypting the keys on disk e.g. in combination with an entered pin to unlock the phone.
+
+*Note, this again not a exhaustive list on all the options for managing cryptographic keys on the phone. More work is required here.*
+
+### How to authenticate the user 
+
+**Assumption**: The coordinator api does not require any authentication - however we might want to ensure that it is only used from the 10101 app.
+
+The coordinator api should though only be exposed through HTTPS.
+
+## Backup and restore (remember punish transactions etc.)
+
+The user needs to be able to backup not only his keys but also his channel (dlc) data so that she can restore it on another phone. 
+
+- **Option 1 *Closing and reopening all channels and positions***: A costly, but straight forward simplified backup process. However also costly (as on chain transactions are required) and probably not feasible as the market price for a position will shift and might not get matched when reopening.   
+- **Option 2 *Backup to google cloud, icloud, nas backup, SW3 Bucket, Zip File, etc.***: Also quite straight forward, but might produce race conditions (complexities) as the apps state could live on multiple phones.
```

Comment:
Most platforms explicitly say that using the same backup on different devies at the same time is not supported and might result in a loss of funds.